app/src/main/java/net/gsantner/opoc/ui/SearchOrCustomTextDialog.java

```diff
@@ -397,7 +352,26 @@ public void beforeTextChanged(final CharSequence arg0, final int arg1, final int
             return true;
         };
 
-        // Click listener set to select or activate as appropriate
+        // Helper function to set ok button enabled and count
+        final Button okButton = dialog.getButton(Dialog.BUTTON_POSITIVE);
+        final String okWithCount = activity.getString(dopt.okButtonText) + " (%d)";
+        final Callback.a0 setOkButtonState = () -> {
+            final String searchText = dopt.isSearchEnabled ? searchEditText.getText().toString() : null;
+            final boolean positionCallbackPossible = dopt.positionCallback != null && !listAdapter._selectedItems.isEmpty();
+            final boolean textCallbackPossible = dopt.callback != null && !TextUtils.isEmpty(searchText);
+            okButton.setEnabled(positionCallbackPossible || textCallbackPossible);
```

Comment:
I have added logic to grey out the ok button if there is no available action. I can remove this if required.

Reply:
suggested:

okWithCount = activity.getString(dopt.okButtonText) + (dopt.isMulti ? " (%d) : "");

later: okButton.setText(dopt.okButtonText.replace("%d", size()));